patches/tModLoader/Terraria/ModLoader/NPCLoader.cs

```diff
@@ -1218,5 +1218,22 @@ internal static void DrawBehind(NPC npc, int index)
 				}
 			}
 		}
+
+		private delegate bool DelegateNeedSaving(NPC npc);
+		private static HookList HookNeedSaving = AddHook<DelegateNeedSaving>(g => g.NeedSaving);
+
+		public static bool SavesAndLoads(NPC npc) {
+			if (npc.townNPC && npc.type != NPCID.TravellingMerchant)
+				return false;
```

Comment:
You sure about this? Pretty sure modded townNPCs should be able to save and load

Reply:
Sorry if the name is a bit misleading, SavesAndLoad is only for saving outside of town NPCs, they have their own saving not related to this one